src/Client/ConnectionString.cpp

```diff
@@ -0,0 +1,218 @@
+#include "ConnectionString.h"
+
+#include <Common/Exception.h>
+#include <Poco/Exception.h>
+#include <Poco/URI.h>
+
+#include <array>
+#include <iostream>
+#include <string>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+}
+
+namespace
+{
+
+using namespace std::string_literals;
+using namespace std::literals::string_view_literals;
+
+constexpr auto CONNECTION_URI_SCHEME = "clickhouse:"sv;
+
+void uriDecode(std::string & uri_encoded_string, bool plus_as_space)
```

Comment:
Let's simply return the un-encoded string (no in-out param).

Reply:
Done